streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -436,19 +468,20 @@ public String toString() {
         final Sensor taskCommitTimeSensor;
 
 
-        public TaskMetrics(StreamsMetrics metrics) {
-            String name = id.toString();
+        public TaskMetrics(final StreamsMetrics metrics) {
+            final String name = id.toString();
             this.metrics = (StreamsMetricsImpl) metrics;
-            this.taskCommitTimeSensor = metrics.addLatencyAndThroughputSensor("task", name, "commit", Sensor.RecordingLevel.DEBUG, "streams-task-id", name);
+            taskCommitTimeSensor = metrics.addLatencyAndThroughputSensor("task", name, "commit", Sensor.RecordingLevel.DEBUG, "streams-task-id", name);
         }
 
-        public void removeAllSensors() {
+        void removeAllSensors() {
             metrics.removeSensor(taskCommitTimeSensor);
         }
     }
 
     @Override
-    public void flushState() {
+    protected void flushState() {
```

Comment:
Since now stream thread does not directly call this function anymore, could it be `no modifier`?